text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/builtin/BuiltInTags.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2022 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.text.minimessage.tag.builtin;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Stream;
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.kyori.adventure.text.format.TextDecoration;
+import net.kyori.adventure.text.minimessage.tag.resolver.TagResolver;
+
+/**
+ * Tag types distributed with MiniMessage.
+ *
+ * <p>All built-in types are included in the default tag resolver.</p>
+ *
+ * @since 4.10.0
+ */
+public final class BuiltInTags {
+  private BuiltInTags() {
+  }
+
+  private static final TagResolver DECORATION = Stream.concat(TextDecoration.NAMES.keyToValue().entrySet().stream(), DecorationTag.DECORATION_ALIASES.entrySet().stream())
+    .flatMap(entry -> {
+      return Stream.of(
```

Comment:
not 100% sure if I like having tags declared inline, having every single one as a class gave a nice overview and would keep this class as a dum registry

Reply:
hm, an alternative could be, for each tag:

```java
final class ClickTag {
  static final String CLICK = "click";
  static final TagResolver RESOLVER =TagResolver.resolver("click", ClickTag::create);
}
```

which would break things up more?

I do think the structure of tag classes will change some once we fix up serialization though -- I want serialization to be associated with each tag resolver, rather than one god class. Since serialization fixes are still part of my goals for 4.10.0, are you fine with pushing some of that discussion off until I have a serialization PR ready to go?

Reply:
yep, fine to move that out of this PR, its big enough as is. 
totally agree with having serialization and deserialization in the same tag class, that way it should be organized. 

(originally, minimessage was just one class, moving out to transformations (new tags) has been a huge improvement, its just that the serializer didn't get that treatment yet)